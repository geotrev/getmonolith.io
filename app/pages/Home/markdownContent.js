import React from "react"
import Markdown from "react-markdown"
import Loadable from "react-loadable"

import installNpm from "./installNpm.md"
import installAssets from "./installAssets.md"

const loader = () => import("./Badges")
const loading = () => <span className="is-visually-hidden">Loading badges</span>
export const StatusBadges = Loadable({ loader, loading })

export const InstallNpm = () => <Markdown source={installNpm} escapeHtml={false} />
export const InstallAssets = () => <Markdown source={installAssets} escapeHtml={false} />
