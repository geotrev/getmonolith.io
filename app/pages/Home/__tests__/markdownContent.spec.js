import React from "react"
import { StatusBadges, InstallNpm, InstallAssets } from "../markdownContent"

jest.mock("../markdown/installAssets.md", () => "install assets!")
jest.mock("../markdown/installNpm.md", () => "install npm!")

describe("<StatusBadges />", () => {
  it("renders loader ", () => {
    const wrapper = shallow(<StatusBadges />)
    expect(wrapper).toMatchSnapshot()
  })

  it("renders component once resolved", async () => {
    const wrapper = await shallow(<StatusBadges />)
    expect(wrapper).toMatchSnapshot()
  })
})

describe("<InstallNpm />", () => {
  it("renders", () => {
    const wrapper = mount(<InstallNpm />)
    expect(wrapper).toMatchSnapshot()
  })
})

describe("<InstallAssets />", () => {
  it("renders", () => {
    const wrapper = mount(<InstallAssets />)
    expect(wrapper).toMatchSnapshot()
  })
})
