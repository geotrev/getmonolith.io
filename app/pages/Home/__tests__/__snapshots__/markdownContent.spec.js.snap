// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InstallAssets /> renders 1`] = `
<InstallAssets>
  <ReactMarkdown
    astPlugins={Array []}
    escapeHtml={false}
    parserOptions={Object {}}
    plugins={Array []}
    rawSourcePos={false}
    renderers={Object {}}
    skipHtml={false}
    source="install npm!"
    sourcePos={false}
    transformLinkUri={[Function]}
  >
    <Root
      key="root-1-1-0"
    >
      <p
        key="paragraph-1-1-0"
      >
        <TextRenderer
          key="text-1-1-0"
          nodeKey="text-1-1-0"
          value="install npm!"
        >
          install npm!
        </TextRenderer>
      </p>
    </Root>
  </ReactMarkdown>
</InstallAssets>
`;

exports[`<InstallNpm /> renders 1`] = `
<InstallNpm>
  <ReactMarkdown
    astPlugins={Array []}
    escapeHtml={false}
    parserOptions={Object {}}
    plugins={Array []}
    rawSourcePos={false}
    renderers={Object {}}
    skipHtml={false}
    source="install npm!"
    sourcePos={false}
    transformLinkUri={[Function]}
  >
    <Root
      key="root-1-1-0"
    >
      <p
        key="paragraph-1-1-0"
      >
        <TextRenderer
          key="text-1-1-0"
          nodeKey="text-1-1-0"
          value="install npm!"
        >
          install npm!
        </TextRenderer>
      </p>
    </Root>
  </ReactMarkdown>
</InstallNpm>
`;

exports[`<StatusBadges /> renders component once resolved 1`] = `<Badges />`;

exports[`<StatusBadges /> renders loader  1`] = `
<loading
  error={null}
  isLoading={true}
  pastDelay={false}
  retry={[Function]}
  timedOut={false}
/>
`;
